a3_finance/overrides/salary_slip.py

```
@@ -131,34 +132,87 @@ def calculate_exgratia(doc, method):
+#     if not doc.gross_pay:
+#         return  # If gross pay is not available, exit
+#     if doc.gross_pay ==0:
+#         frappe.db.set_value("Salary Slip", doc.name, "custom_professional_tax", slab.tax_amount)
+#     # Get active Profession Tax document
+#     profession_tax = frappe.get_value("Profession Tax", {"is_active": 1}, "name")
+#     if not profession_tax:
+#         frappe.log_error("No active Profession Tax document found", "Professional Tax Calculation")
+#         return
+
+#     profession_tax_doc = frappe.get_doc("Profession Tax", profession_tax)
+    
```

Ruff (0.12.2)

Blank line contains whitespace

Remove whitespace from blank line

(W293)